pkg/storage/snapshot/snapshot.go

```diff
@@ -728,21 +721,32 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		} else {
 			updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Source does not exist"))
 		}
-		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionFalse, "Not ready"))
+
 		if vmSnapshotDeleting(vmSnapshotCpy) {
 			vmSnapshotCpy.Status.Phase = snapshotv1.Deleting
 			updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "VM snapshot is deleting"))
-			updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionFalse, "VM snapshot is deleting"))
 		}
-	} else if VmSnapshotReady(vmSnapshotCpy) {
-		vmSnapshotCpy.Status.Phase = snapshotv1.Succeeded
-		updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Operation complete"))
-		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionTrue, "Operation complete"))
-		updateSnapshotSnapshotableVolumes(vmSnapshotCpy, content)
+	}
+
+	if VmSnapshotReady(vmSnapshotCpy) {
```

Comment:
I think the ready check should be before all the phases checks to have a default ready/not ready reasons but that each other phase can add its own explanation if it changed the ready condition like:
`VM snapshot is deleting`, `Operation complete`, `Operation failed`,`Error` etc..

Reply:
in order to reduce updates, `updateSnapshotCondition` will only update the condition message if the condition value is changed so have to do the opposite